src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,37 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.AgeRestrictionException;
+import core.basesyntax.exceptions.LoginTooShortException;
+import core.basesyntax.exceptions.PasswordTooShortException;
+import core.basesyntax.exceptions.UserAlreadyExistException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int LOGIN_MIN_LENGTH = 6;
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int AGE_MINIMAL_LIMIT = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new IllegalArgumentException("User can't be null!");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new UserAlreadyExistException("User with same login already exist!");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < LOGIN_MIN_LENGTH) {
+            throw new LoginTooShortException("Login should be more then six characters!");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < PASSWORD_MIN_LENGTH) {
+            throw new PasswordTooShortException("Login should be more then six characters!");
+        }
+        if (user.getAge() == null || user.getAge() < AGE_MINIMAL_LIMIT) {
+            throw new AgeRestrictionException(
+                    "User with age less then 18 or null, can't be created!"
```

Comment:
The exception message is not entirely informative. It would be clearer if it stated the minimum age requirement, such as 'User must be at least 18 years old.'